Python/Product/Debugger/Debugger/DebugEngine/AD7Engine.cs

```diff
@@ -561,7 +562,7 @@ public sealed class AD7Engine : IDebugEngine2, IDebugEngineLaunch2, IDebugProgra
             _breakOnException.Clear();
             _defaultBreakOnExceptionMode = (int)enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_UNCAUGHT;
 
-            SetExceptionInfo(_defaultBreakOnExceptionMode, _breakOnException);
+            TaskHelpers.RunSynchronouslyOnUIThread(ct => SetExceptionInfoAsync(_defaultBreakOnExceptionMode, _breakOnException, ct));
```

Comment:
It's about using JoinableTaskFactory to block until the asynchronous call is done. The TaskHelpers require you to be on the UI thread already.  The name can be misleading, thinking that it will do the switch to UI thread. It's just what they were named in RTVS.

---
In reply to: [108532812](https://github.com/Microsoft/PTVS/pull/2323#discussion_r108532812) [](ancestors = 108532812)